src/stores/room-list/filters/SpaceFilterCondition.ts

```diff
@@ -34,10 +34,11 @@ export class SpaceFilterCondition extends EventEmitter implements IFilterConditi
     private roomIds = new Set<string>();
     private userIds = new Set<string>();
     private showPeopleInSpace = true;
+    private showIndirectRoomsInSpace = true;
     private space: SpaceKey = MetaSpace.Home;
 
     public isVisible(room: Room): boolean {
-        return SpaceStore.instance.isRoomInSpace(this.space, room.roomId);
+        return SpaceStore.instance.isRoomInSpace(this.space, room.roomId, this.showIndirectRoomsInSpace);
     }
 
     private onStoreUpdate = async (forceUpdate = false): Promise<void> => {
```

Comment:
Looks like `this.roomIds` and `this.userIds` don't respect `this.showIndirectRoomsInSpace`